### PR TITLE
fix/update react 17

### PR DESCRIPTION
**Pull request by nicolastoulemont**
This update the package to react 17 and resolve the opening of the popup when running React v17.

I've run the React codemods to prefix the legacy methods and update the dependency that was the root cause of the issue.

The large diff is caused by the fact that for some reason the bundle is included in the git history and some change in the yarn.lock

### CODE REVIEW

**Review by soupette — CHANGES_REQUESTED** [review]
Can you add the bundle.js and bundle.js.map to the .gitignore?

Also, why is there a diff in the fontawesome files?

Other than that good job you are a lifesaver on this one!

**Comment by nicolastoulemont** [comment]
> Also, why is there a diff in the fontawesome files?

They're outputted by the `build:example` command which I re-ran to ensure that they're up to date. Not sure why they changed 🤔 
They're not included in the pkg anyway (only the lib folder is released)

File: .gitignore [hunk 1]
```diff
@@ -32,4 +32,7 @@ node_modules
 .idea
 
 # Mac OS X
-.DS_Store
\ No newline at end of file
+.DS_Store
+
+examples/bundle.js
```
Comment (soupette) [inline .gitignore:37]: Can you delete the bundle.js and bundle.js.map too so we don't have them in the history?

Comment (nicolastoulemont) [reply .gitignore:37]: They've been deleted as well

**Review by soupette — APPROVED** [review]
LGTM

**Pull request merged**